### PR TITLE
Merge adding per core support

### PR DESCRIPTION
**Pull request by vjsamuel**
https://github.com/elastic/topbeat/issues/40

Kindly review this page for the issue of not reporting per core cpu metrics.

### CODE REVIEW

**Comment by ruflin** [comment]
@vjsamuel Thanks for the addition. It some seems to break the system tests on Windows: https://ci.appveyor.com/project/elastic-beats/topbeat/build/48#L91 Not sure if some of this data is not available on Windows or in a different way?

**Comment by vjsamuel** [comment]
Is there a way by which I could debug this?

**Comment by vjsamuel** [comment]
https://github.com/elastic/gosigar/pull/5

to fix the issue of panic() causing the program to fail on windows.

**Comment by vjsamuel** [comment]
How long does it normally take for patches to get merged, packaged and shipped out?

**Comment by monicasarbu** [comment]
I merged the PR that you created in elastic/gosigar. In order to get those changes into Topbeat, you need to update Godeps:

       godep update github.com/elastic/gosigar

Please push the result, so Appveyor can run the tests on Windows.

It's too late for this change to make it in 1.0.0 version which we plan to release early next week. So, this fix will be in 1.1.0 release which we target middle of January. In the meantime you can use the nightly builds from: https://beats-nightlies.s3.amazonaws.com/index.html?prefix=topbeat/

**Comment by vjsamuel** [comment]
i ve done the needful

**Comment by tsg** [comment]
@vjsamuel Tests didn't run because of a merge conflict in docs :-(. Can you rebase master, please?

File: topbeat.go [hunk 1]
```diff
@@ -342,6 +355,32 @@ func (t *Topbeat) addCpuPercentage(t2 *CpuTimes) {
...
+func (t *Topbeat) addCpuPercentageList(t2 []CpuTimes) {
+
+	t1 := t.lastCpuTimesList
+
+	if t1 != nil && t2 != nil && len(t1) == len(t2) {
+
+		for i := 0; i < len(t1); i++ {
+			all_delta := t2[i].sum() - t1[i].sum()
+
+			calculate := func(field2 uint64, field1 uint64) float64 {
```
Comment (tsg) [inline topbeat.go]: I think this function "definition" would be better moved up out of the loop. Maybe the optimizer can do it, not sure, but I think the code would be cleaner that way anyway.

**Pull request merged**